spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/TypeToSparkType.java

```diff
@@ -59,8 +61,11 @@ public DataType struct(Types.StructType struct, List<DataType> fieldResults) {
     for (int i = 0; i < fields.size(); i += 1) {
       Types.NestedField field = fields.get(i);
       DataType type = fieldResults.get(i);
-      StructField sparkField = StructField.apply(
-          field.name(), type, field.isOptional(), Metadata.empty());
+      Metadata metadata = Metadata.empty();
+      if (MetadataColumns.isMetadataColumn(field.name())) {
+        metadata = new MetadataBuilder().putBoolean(MetadataColumns.METADATA_COL_ATTR_KEY, true).build();
+      }
```

Comment:
Could this be refactored into a separate method? I think that would be cleaner. And, this should identify metadata columns using column IDs, since they are available in Iceberg schemas.

Reply:
makes sense, made it consistent to 3.3 review comments as well.